cmake/match.cmake

```diff
@@ -6,32 +6,50 @@
 # with a known good match file.
 #
 
-if(NOT DEFINED TEST_FILE)
-    message(FATAL_ERROR "TEST_FILE needs to be defined")
+# Process variables passed to the script
+if(NOT DEFINED MODE)
+    message(FATAL_ERROR "MODE needs to be defined. Possible values: 'binary', 'file'")
```

Comment:
I think really the difference is what we match against, some specified file or stdout, right? If so, I think it would be better to call these modes 'stdout' and 'file'.